VoxelisedSimulation/SubScripts/ComputePoissonDataCorrections.sh

```diff
@@ -0,0 +1,164 @@
+#	10: scatter_recon_num_subsets				- The number of reconstruction subsets [Default: 18]
+
+set -e # exit on error
+trap "echo ERROR in $0" ERR
+
+if [[ $# == 7 || $# == 10 ]]; then
+	## Required arguments
+	#  Output filename
+	OutputMultiplicativeFactorsFilename=$1
+	OutputAdditiveSinogramFilename=$2
+	# Input filenames
+	sino_input=$3
+	DelayedData=$4
+	NORM=$5
+	atnimg=$6
+	AttenIsGATE=$7
+else
+	echo "Usage: ${0} OutputMultiplicativeFactorsFilename OutputAdditiveSinogramFilename MeasuredData[sino_input] DelayedData NormalisationSinogram[NormSino] AttenuationImage[atnimg] AttenuationIsGateOutput[AttenIsGATE] Optional:[num_scat_iters scatter_recon_num_subiterations scatter_recon_num_subsets]"
+	echo "The final option (AttenIsGATE) is required because GATE images are inverted in z axis (no idea why...)"
+	exit 1
+fi
+
+if [[ $# == 10 ]]; then
+	num_scat_iters=$8
+	scatter_recon_num_subiterations=$9
+	scatter_recon_num_subsets=${10}
+else
+	## Default values
+	echo "Setting optional arguments to default"
+	num_scat_iters=5
+	scatter_recon_num_subiterations=18
+	scatter_recon_num_subsets=18
+fi
+
+
+## Echo the script arguments for debugging
+echo "  =========================  "
+echo "Compute Poisson Data Correction script will use the following configuration:"
+echo "    OutputMultiplicativeFactorsFilename = ${OutputMultiplicativeFactorsFilename}"
+echo "    OutputAdditiveSinogramFilename = ${OutputAdditiveSinogramFilename}"
+echo "    CoincidenceData[sino_input] = ${sino_input}"
+echo "    DelayedData = ${DelayedData}"
+echo "    NormalisationSinogram[NormSino] = ${NormSino}"
+echo "    AttenuationImage[atnimg] = ${atnimg}"
+echo "    AttenuationIsGateOutput[AttenIsGATE] = ${AttenIsGATE}"
+echo "Optional variables:"
+echo "    num_scat_iters = ${num_scat_iters}"
+echo "    scatter_recon_num_subiterations = ${scatter_recon_num_subiterations}"
+echo "    scatter_recon_num_subsets = ${scatter_recon_num_subsets}"
+echo "  =========================  "
+
+
+## SETUP of variables: No need to change stuff here, setup for exports and files will be deleted with cleanup
+acf3d=my_attenuation_coefficients.hs
+scatter_prefix=my_scatter
+total_additive_prefix=my_total_additive
+mask_image=my_mask
+mask_projdata_filename=my_sino_mask
+randoms3d_prefix=my_randoms
+randoms3d=${randoms3d_prefix}.hs
+cleanup=1 ## At the end of the script delete all files except $OutputAdditiveSinogramFilename and $multfactors
+
+## Gets the scatter par(dir) from the STIR install path (STIR examples are installed with latest versions of STIR)
+STIR_install_dir=$(dirname $(dirname $(command -v estimate_scatter)))
+scatter_pardir="$STIR_install_dir/share/doc/stir-5.0/examples/samples/scatter_estimation_par_files"
+scatter_par="${scatter_pardir}/scatter_estimation.par"
+
+
+## Estimate the randoms from delays
+echo "====="
+echo "Computing contribution due to randoms from DelayedData"
+sh SubScripts/EstimateRandomsFromDelayed.sh ${randoms3d} ${DelayedData}
```

Comment:
As that script uses `trap "echo ERROR in $0" ERR` it is actually a `bash` script, so use `bash` here and change the first line in the script.

(In some systems, `sh` is the same as `bash`, but not all)

Reply:
Agreed, I should probably remove these at some point to make it general with shell